cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/connect/class-api.php

```diff
@@ -511,13 +511,14 @@ public function create_local_copy( $file ) {
 	/**
 	 * Expicit update of an asset.
 	 *
-	 * @param array $args Array of options to update.
+	 * @param array  $args Array of options to update.
+	 * @param string $type The type of endpoint that's being called: 'video' or 'image'.
 	 *
 	 * @return array|\WP_Error
 	 */
-	public function explicit( $args ) {
+	public function explicit( $args, $type = 'image' ) {
 
-		$url  = $this->url( 'image', 'explicit', true );
+		$url  = $this->url( $type, 'explicit', true );
 		$args = $this->clean_args( $args );
```

Comment:
@dugajean `eager` param needs to be a string base not an array, so before `clean_args()` you'll need to do something like :
```php
		if ( ! empty( $args['eager'] ) ) {
			$args['eager'] = self::generate_transformation_string( $args['eager'], $type );
		}
```